src/solve.jl

```diff
@@ -28,12 +28,12 @@ end
 
 for Sa in [(2,2), (3,3)]  # not needed for Sa = (1, 1);
     @eval begin
-        @inline function solve(::Size{$Sa}, ::Size{Sb}, a::StaticMatrix{<:Any, <:Any, Ta}, b::StaticMatrix{<:Any, <:Any, Tb}) where {Sb, Ta, Tb}
+        @inline function _solve(::Size{$Sa}, ::Size{Sb}, a::StaticMatrix{<:Any, <:Any, Ta}, b::StaticMatrix{<:Any, <:Any, Tb}) where {Sb, Ta, Tb}
             d = det(a)
             T = typeof((one(Ta)*zero(Tb) + one(Ta)*zero(Tb))/d)
             c = similar(b, T)
             for col = 1:Sb[2]
-                @inbounds c[:, col] = solve(Size($Sa), Size($Sa[1],), a, b[:, col])
+                @inbounds c[:, col] = _solve(Size($Sa), Size($Sa[1],), a, b[:, col])
```

Comment:
Not sure here. Does `b[: col]` allocate if `b` is a `MArray`?

Reply:
No, this is one of the luxuries of the julia-1.0 compiler: a temporary `MArray` which doesn't escape the function doesn't result in any allocations. I still can't quite believe it myself :-)

Reply:
Nice! LGTM